src/Namecheap/Helper/NamecheapApi.php

```diff
@@ -0,0 +1,602 @@
+    {
+        $params = [
+            'command' => 'namecheap.domains.dns.getList',
+            'SLD'     => $sld,
+            'TLD'     => $tld,
+        ];
+
+        return $this->makeRequest($params)->DomainDNSGetListResult;
+    }
+
+    /**
+     * @param  string  $domainName
+     *
+     * @return bool
+     */
+    public function getRegistrarLockStatus(string $domainName): bool
+    {
+        $params = [
+            'command'    => 'namecheap.domains.getRegistrarLock',
+            'DomainName' => $domainName,
+        ];
+
+        $tld = Utils::getTld($domainName);
+
+        if (in_array($tld, self::NO_LOCK_TLDS)) {
+            return false;
+        }
+
+        $response = $this->makeRequest($params);
+        $lockStatus = (string) $response->DomainGetRegistrarLockResult->attributes()->RegistrarLockStatus;
+
+        return $lockStatus === "true";
+    }
+
+    /**
+     * @param  string  $domainName
+     *
+     * @return SimpleXMLElement
+     */
+    private function getContacts(string $domainName): SimpleXMLElement
+    {
+        $params = [
+            'command'    => 'namecheap.domains.getContacts',
+            'DomainName' => $domainName,
+        ];
+
+        return $this->makeRequest($params)->DomainContactsResult;
+    }
+
+    /**
+     * @param  string  $domainName
+     *
+     * @return array|null
+     */
+    public function getDomainTransferOrders(string $domainName): ?array
+    {
+        $params = [
+            'command'    => 'namecheap.domains.transfer.getlist',
+            'SearchTerm' => $domainName,
+            'ListType'   => 'INPROGRESS',
+        ];
+
+        $response = $this->makeRequest($params);
+        $orderCount = (int) $response->Paging->TotalItems;
+
+        $orders = [];
+
+        if ($orderCount > 0) {
+            foreach ($response->TransferGetListResult->children() as $order) {
+                $orders[] = [
+                    'orderId'  => (string) $order->attributes()->OrderID,
+                    'status'   => (string) $order->attributes()->Status,
+                    'statusId' => (int) $order->attributes()->StatusID,
+                    'date'     => Utils::formatDate((string) $order->attributes()->StatusDate),
+                ];
+            }
+
+            if (count($orders) > 0) {
+                return $orders;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Send request and return the response.
+     *
+     * @param  array  $params
+     *
+     * @return SimpleXMLElement
+     *
+     * @throws ProvisionFunctionError
+     */
+    public function makeRequest(array $params): SimpleXMLElement
+    {
+        // Prepare command params
+        $params = array_merge([
+            'ApiUser'  => $this->configuration->username,
+            'ApiKey'   => $this->configuration->api_token,
+            'UserName' => $this->configuration->username,
+            'ClientIp' => request()->server('SERVER_ADDR'),
```

Comment:
This is a really interesting requirement from the Namecheap API. Unfortunately there's no 100% reliable way of determining the system IP in all execution environments (e.g., during background processes or in "non-traditional" server configurations / containers etc), this approach will not work.

So, I'll work on some changes to the base library to inject environment values / metadata that can contain the system IP amongst other things, and then will update this code to allow the IP to be injected to the method or as a class property instead of relying on global state.